subsys/pm/pm.c

```diff
@@ -26,6 +26,10 @@ LOG_MODULE_REGISTER(pm, CONFIG_PM_LOG_LEVEL);
 #define CURRENT_CPU \
 	(COND_CODE_1(CONFIG_SMP, (arch_curr_cpu()->id), (_current_cpu->id)))
 
+#define PM_WORKQ_STACK_SIZE 512
```

Comment:
Should we consider making this a Kconfig option to make it user-configurable?

Reply:
I like the idea of the work queue, but mix it with notifier is probably not the best solution.

Notifier callback gives you the action that triggered it (entry/leave) and an user context, also, when the work is defined the notification is not called.

Reply:
Agreed. My current approach would be to have `pm_system_resume()` only schedule the actual resume logic in the workqueue and then exit. With that running on wakeup in an ISR or idle thread (depending on platform), the workqueue should then immediately start execution once scheduled.